### PR TITLE
fix(sumologicexporter): deduplicate error messages

### PR DESCRIPTION
**Pull request by astencel-sumo**
Refs: SUMO-185930

### CODE REVIEW

File: pkg/exporter/sumologicexporter/exporter.go [hunk 1]
```diff
@@ -440,13 +441,32 @@ func (se *sumologicexporter) pushMetricsData(ctx context.Context, md pdata.Metri
...
 
+// removeDuplicates removes duplicate errors from a collection of errors.
+// Duplicate errors are errors that have the same message returned by `error.Error()`.
+func removeDuplicates(errs []error) []error {
+	if len(errs) < 2 {
+		return errs
+	}
+
+	var uniqueList []error
+	uniqueSet := make(map[string]bool)
```
Comment (pmalek-sumo) [inline pkg/exporter/sumologicexporter/exporter.go:460]:
```suggestion
	uniqueSet := make(map[string]struct{})
```

and then use not the value but the second value returned by the map check

**Review by pmalek-sumo — COMMENTED** [review]
In general I'm sort of against this change (but I can understand its benefits like cleaner log messages) because it hides away the scale of the problem i.e. the number of errors that occurred.

This way it may be harder to correlate requests sent to errors received.

**Comment by sumo-drosiek** [comment]
@pmalek-sumo so maybe we should add counter to the errors, so we won't loose this information

**Comment by swiatekm-sumo** [comment]
> @pmalek-sumo so maybe we should add counter to the errors, so we won't loose this information

I like this idea. We can use a format like `{error_message} (x{error_count})`, which I think is both clear and compact.

File: pkg/exporter/sumologicexporter/exporter.go (hunk 1 above)
Comment (astencel-sumo) [reply pkg/exporter/sumologicexporter/exporter.go:460]: Done

**Comment by astencel-sumo** [comment]
> > @pmalek-sumo so maybe we should add counter to the errors, so we won't loose this information
> 
> I like this idea. We can use a format like `{error_message} (x{error_count})`, which I think is both clear and compact.

I agree, will implement this.

**Comment by astencel-sumo** [comment]
Implemented counting errors in https://github.com/SumoLogic/sumologic-otel-collector/pull/446/commits/704dc75d507e29a1555c4b64782d1acf85c791c9, please review @pmalek-sumo @sumo-drosiek @swiatekm-sumo

File: pkg/exporter/sumologicexporter/deduplicate_errors.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	errorsWithCounts := []errorWithCount{}
+	for _, err := range errs {
+		found := false
+		for i := range errorsWithCounts {
+			if errorsWithCounts[i].err.Error() == err.Error() {
+				found = true
+				errorsWithCounts[i].count += 1
+				break
+			}
+		}
```
Comment (sumo-drosiek) [inline pkg/exporter/sumologicexporter/deduplicate_errors.go:37]: wouldn't be better to use map instead of list?

Comment (astencel-sumo) [reply pkg/exporter/sumologicexporter/deduplicate_errors.go:37]: I want to keep the order of the errors. With a map, I'm losing the ordering.

**Review by pmalek-sumo — APPROVED** [review]

**Pull request merged**